compiler/semantic/expr.go

```diff
@@ -519,64 +519,82 @@ func (a *analyzer) semAssignments(assignments []ast.Assignment, summarize bool)
 	return out, nil
 }
 
-func (a *analyzer) semAssignment(assign ast.Assignment, summarize bool) (dag.Assignment, error) {
-	rhs, err := a.semExpr(assign.RHS)
+func (a *analyzer) semAssignment(e ast.Assignment) (dag.Assignment, error) {
+	rhs, err := a.semExpr(e.RHS)
 	if err != nil {
-		return dag.Assignment{}, fmt.Errorf("rhs of assignment expression: %w", err)
-	}
-	if _, ok := rhs.(*dag.Agg); ok {
-		summarize = true
+		return dag.Assignment{}, err
```

Comment:
Maybe stick with
```go
 		return dag.Assignment{}, fmt.Errorf("rhs of assignment expression: %w", err)
```
(and likewise for an LHS error) since this seems like it provides useful context?

Reply:
Is lhs/rhs a user friendly abbreviation or should it be left-hand side / right-hand side?

Reply:
I'd advocate for the more verbose "left-hand side / right-hand side".